fairlearn/metrics/_multi_curve_plot.py

```diff
@@ -0,0 +1,274 @@
+from typing import Callable, Union
+from numpy import amax, amin, array
+from sklearn.utils.validation import check_array
+from numpy import zeros
+
+
+def plot_model_comparison(
+    *,
+    x_axis_metric: Callable[..., Union[float, int]],
+    y_axis_metric: Callable[..., Union[float, int]],
+    y_true,
+    y_preds,
+    sensitive_features,
+    ax=None,
+    axis_labels=True,
+    point_labels=None,
+    model_kwargs=None,
+    legend=False,
+    legend_kwargs={},
+    plot=True,
+    **kwargs,
+):
+    r"""
+    Plot a model comparison.
+
+    Parameters
+    ----------
+    x_axis_metric : Callable
+        The (aggregating) metric function for the x-axis
+        The passed metric function must take `y_true, y_pred`, and optionally `sensitive_features`.
+        If the metric is grouped, it must aggregate results. For instance, use
+        `make_derived_metric(metric=balanced_accuracy_score, transform='group_min')`
+        to aggregate the `balanced_accuracy_score`.
+
+    y_axis_metric : Callable
+        The (aggregating) metric function for the y-axis, similar to x_axis_metric.
+        The passed metric function must take `y_true, y_pred`, and optionally `sensitive_features`.
+        If the metric is grouped, it must aggregate results.
+
+    y_true : List, pandas.Series, numpy.ndarray, pandas.DataFrame
+        The ground-truth labels (for classification) or target values (for regression).
+
+    y_preds : array-like
+        An array-like containing predictions per model. Hence, predictions of
+        model :code:`i` should be in :code:`y_preds[i]`.
+
+    sensitive_features : List, pandas.Series, dict of 1d arrays, numpy.ndarray, pandas.DataFrame, optional
+        The sensitive features which should be used to create the subgroups.
+        At least one sensitive feature must be provided.
+        All names (whether on pandas objects or dictionary keys) must be strings.
+        We also forbid DataFrames with column names of ``None``.
+        For cases where no names are provided
+        we generate names ``sensitive_feature_[n]``.
+
+    ax : matplotlib.axes.Axes, optional
+        If supplied, the scatter plot is drawn on this Axes object.
+        Else, a new figure with Axes is created.
+
+    axis_labels : bool
+        If true, add the names of x and y axis metrics
+
+    point_labels : list
+        Add textual label :code:`point_labels[i]` to the position
+        of model :code:`i`.
+
+    model_kwargs : list[dict], optional
+        For a model at index :code:`i` (same order as in :code:`y_preds`),
+        :code:`model_kwargs[i]` are passed along to matplotlib's scatter
+        plotting. If models have the same `model_kwargs`, they are passed
+        in one call to :py:func:`Axes.scatter`, so they become one entry in
+        the legend.
+
+    legend : bool
+        If True, add a legend. Must set
+        :code:`model_kwargs[i]['label']` for every prediction :code:`i`.
+
+    legend_kwargs : dict
```

Comment:
I don't think we need this one... this functionality should be handled on the outside--see my comment under `model_kwargs`.

Reply:
Resurfacing this comment. Can we remove `legend_kwargs` for simplicity?